pandas/core/indexes/multi.py

```diff
@@ -1126,27 +1126,39 @@ def _to_safe_for_reshape(self):
         """ convert to object if we are a categorical """
         return self.set_levels([i._to_safe_for_reshape() for i in self.levels])
 
-    def to_frame(self, index=True):
+    def to_frame(self, index=True, name=None):
         """
         Create a DataFrame with the levels of the MultiIndex as columns.
 
-        .. versionadded:: 0.20.0
+        .. versionadded:: 0.24.0
 
         Parameters
         ----------
         index : boolean, default True
             Set the index of the returned DataFrame as the original MultiIndex.
 
+        name : list / sequence of strings or None
+            The passed names should substitute for the series names (if it has
+            one).
+
         Returns
         -------
         DataFrame : a DataFrame containing the original MultiIndex data.
         """
 
         from pandas import DataFrame
+        if name is not None:
+            if len(name) != len(self.levels):
```

Comment:
Should we protect against people accidentally using a single string here, rather than a list of strings? You can use `is_list_like` to check I think.